src/rgw/rgw_rados.cc

```diff
@@ -145,6 +145,33 @@ static bool rgw_obj_to_raw(const RGWZoneGroup& zonegroup, const RGWZoneParams& z
   return rgw_get_obj_data_pool(zonegroup, zone_params, placement_id, obj, &raw_obj->pool);
 }
 
+bool rgw_s3_is_multipart_encrypted(
+  std::map<std::string, ceph::bufferlist>& attrs)
+{
+  auto attr_iter1 = attrs.find(RGW_ATTR_CRYPT_MODE);
+  if (attr_iter1 == attrs.end()) {
+    return false;
+  }
+  auto attr_iter2 = attrs.find(RGW_ATTR_MANIFEST);
+  if (attr_iter2 == attrs.end() || attr_iter2->second.length() == 0) {
+    return false;
+  }
+  RGWObjManifest manifest;
```

Comment:
so the decoded manifest is local to rgw_s3_is_multipart_encrypted?  that's probably ok, but I wonder if having the manifest...manifest at a higher level is helpful, longer run?

Reply:
+1 it looks like `RGWRados::copy_obj()` already has access to a decoded version of the manifest in `astate->manifest`?